src/Couchbase.Extensions.Caching/CouchbaseCache.cs

```diff
@@ -125,22 +115,17 @@ public void Set(string key, byte[] value, DistributedCacheEntryOptions options =
                 throw new ArgumentNullException(nameof(value));
             }
 
-            await Bucket.UpsertAsync(key, value, GetLifetime(options)).ConfigureAwait(false);
+            var collection = await CollectionProvider.GetCollectionAsync().ConfigureAwait(false);
+            await collection.UpsertAsync(key, value, new UpsertOptions().Transcoder(_transcoder).Expiry(GetLifetime(options)))
+                .ConfigureAwait(false);
         }
 
         /// <summary>
         /// Refreshes or "touches" a key updating it's lifetime expiration.
         /// </summary>
         /// <param name="key">The key for the cache item.</param>
-        public void Refresh(string key)
-        {
-            if (key == null)
-            {
-                throw new ArgumentNullException(nameof(key));
-            }
-
-            Bucket.Touch(key, GetLifetime());
-        }
+        public void Refresh(string key) =>
+            RefreshAsync(key).GetAwaiter().GetResult();
```

Comment:
Should we throw NotSupportedException instead? Otherwise perhaps we just document the sync methods as "bad practices".

Reply:
I did add to the docs that we recommend using async, though I could word the language more strongly.

I'm frankly not sure what the best approach is, since it's a bit unclear to me how frequently it's consumed one way or the other. We also have the problem that, theoretically speaking, IDistributedCache may be consumed by other packages that our consumers don't control (just use) which assume that it fully implements the interface.

At this point, I lean towards at least supporting sync, though I do think we should do a separate PR to improve the pattern we use for it.